packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -430,111 +430,150 @@ class TextStyle {
 
   @override
   String toString([String prefix = '']) {
-    final List<String> result = <String>[];
-    result.add('${prefix}inherit: $inherit');
-    if (color != null)
-      result.add('${prefix}color: $color');
-    if (fontFamily != null)
-      result.add('${prefix}family: "$fontFamily"');
-    if (fontSize != null)
-      result.add('${prefix}size: $fontSize');
+    final List<DiagnosticsNode<Object>> properties = <DiagnosticsNode<Object>>[];
+    debugFillProperties(properties, prefix: prefix);
+    // TODO(jacobr): this is a strange toString method for TextStyle.
+    // It would make more sense to specify something along the lines of
+    // TextStyle(property1; property2; property3) as is done elsewhere.
```

Comment:
Fixed and updated test expectations. output for nested cases for text_style are not impact but the toString directly on a text_style now matches the standard conventions. 
This is debug debugFillProperties is called directly for the nested case.